src/variant.rs

```diff
@@ -36,16 +37,54 @@
 //! // `get_str` tries to borrow a string slice.
 //! assert_eq!(hello.get_str(), Some("Hello!"));
 //! assert_eq!(num.get_str(), None);
+//!
+//! // Variant carrying a Variant
+//! let variant = Variant::new_variant(&hello);
+//! let variant = variant.get_variant().unwrap();
+//! assert_eq!(variant.get_str(), Some("Hello!"));
+//!
+//! // Variant carrying an array
+//! let array = [Variant::from("Hello"), Variant::from("there!")];
+//! let variant = Variant::new_array::<&str>(&array);
```

Comment:
What happens if the array does not contain string variants (i.e. if the generic parameter to `new_array()` does not match what's in the array)? Shouldn't this return some kind of error then?

Reply:
The docs in C say

> All items in the array must have the same type, which must be the same as child_type , if given.

So we should probably make that an assertion on our side and document that